### ISSUE TITLE
import turicreate fails on macOS 10.12.6

### ISSUE THREAD

**Issue by rajatarya** [apple/turicreate#256]
Reproduced using turicreate v4.1, does not reproduce with turicreate v4.0.

```
Python 2.7.10 (default, Feb  7 2017, 00:08:15)
Type "copyright", "credits" or "license" for more information.

IPython 5.5.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import turicreate
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-a013a2f69c89> in <module>()
----> 1 import turicreate

/Users/rajat/tmanage/lib/python2.7/site-packages/turicreate/__init__.py in <module>()
     28 from turicreate.version_info import __version__
     29
---> 30 from turicreate.data_structures.sgraph import Vertex, Edge
     31 from turicreate.data_structures.sgraph import SGraph
     32 from turicreate.data_structures.sarray import SArray

/Users/rajat/tmanage/lib/python2.7/site-packages/turicreate/data_structures/__init__.py in <module>()
     16
     17 from . import image
---> 18 from . import sframe
     19 from . import sarray
     20 from . import sgraph

/Users/rajat/tmanage/lib/python2.7/site-packages/turicreate/data_structures/sframe.py in <module>()
     14 from __future__ import division as _
     15 from __future__ import absolute_import as _
---> 16 from ..connect import main as glconnect
     17 from ..cython.cy_flexible_type import infer_type_of_list
     18 from ..cython.context import debug_trace as cython_context

/Users/rajat/tmanage/lib/python2.7/site-packages/turicreate/connect/main.py in <module>()
     11 from __future__ import absolute_import as _
     12
---> 13 from ..cython.cy_unity import UnityGlobalProxy
     14 from ..cython.cy_server import EmbeddedServer
     15

ImportError: dlopen(/Users/rajat/tmanage/lib/python2.7/site-packages/turicreate/cython/cy_unity.so, 2): Symbol not found: _syslog$DARWIN_EXTSN
  Referenced from: /Users/rajat/tmanage/lib/python2.7/site-packages/turicreate/cython/../libunity_shared.dylib (which was built for Mac OS X 10.13)
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/rajat/tmanage/lib/python2.7/site-packages/turicreate/cython/../libunity_shared.dylib

```

**Comment by polymorphia** [comment]
I have the same problem, trying to execute my .py files on a MacOS Sierra 10.12.6 with Python 2.7.10. Is there a workaround for this? Thanks!

**Comment by srikris** [comment]
@polymorphia We are fixing this right away. Your current workaround is to revert to turicreate 4.0.

```
pip uninstall turicreate
pip install turicreate==4.0
```

**Comment by znation** [comment]
Fixed in release 4.1.1.
